### PR TITLE
Ruby 3.0/3.1 support

### PR DESCRIPTION
**Pull request by zendesk-danielmenz**
Update to support ruby 3.0/3.1

Hence it is mostly small changes to support the differences in keyword arguments from 2.7 -> 3.0

### CODE REVIEW

**Review by KJTsanaktsidis — APPROVED** [review]
👍 makes sense to me (but, I did help you write this lol, so maybe someone else should look)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -16,11 +16,18 @@ jobs:
         ruby:
           - 2.6
           - 2.7
+          - 3.0
```
Comment (bquorning) [inline .github/workflows/test.yml:19]:
```suggestion
          - "3.0"
```

Comment (bquorning) [reply .github/workflows/test.yml:19]: The plus side of this mistake is that you have now already tested with Ruby 3.2. So you might as well add it to the list 😅

**Review by bquorning — COMMENTED** [review]
I wonder why `ruby2_keywords` is not necessary, it seems to always be necessary when accepting and forwarding kwargs. @HeyNonster Could you review this too, please?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -16,11 +16,21 @@ jobs:
...
```
Comment (bquorning) [inline .github/workflows/test.yml:19]: Still, `3.0` needs to be quoted, or GHA will test with Ruby 3.2. See https://github.com/zendesk/kasket/actions/runs/4434156832/jobs/7779871418
```suggestion
          - '3.0'
```

**Comment by KJTsanaktsidis** [comment]
> I wonder why ruby2_keywords is not necessary, it seems to always be necessary when accepting and forwarding kwargs

So, in this case, it's not necessary because what we're forwarding _to_ takes kwargs, and we know it. 

We're essentially doing this:

```
def find_by_sql_with_kasket(sql, binds = [], *restargs, **kwargs, &blk)
    find_by_sql_without_kasket(sql, binds, *restarts, **kwargs, &blk)
end

def find_by_sql_without_kasket(sql, binds = [], preparable: nil, &block)
    # implemented in rails (as find_by_sql)
end
```

The signature for `find_by_sql` in Rails has `preparable: nil` in it (i.e. takes kwargs) as far back as I bothered to look (Rails 5.0). It also does _not_ have `ruby2_keywords` on it in Rails. That means, ignoring kasket altogether for a moment, this won't work at all on Ruby 3 anyway:

```
    # Raises ArgumentError on Ruby 3+, because it passes 3 positional arguments to a method expecting 2
    find_by_sql sql, [bind1, bind2], { preparable: true }
    
    # Won't raise an error, but will do entirely the wrong thing - passes the `{preparable: true}` hash as the `binds` parameter.
    find_by_sql sql, { preparable: true }

    # This is what callers _need_ to do
    find_by_sql sql, preparable: true
    find_by_sql sql, **{preparable: true }
```

So, for code to be correct in the absence of kasket, it needs to already be calling `find_by_sql` with kwargs.

Now, let's add kasket into the mix. It injects itself as `find_by_sql`; so, as we established, all its callers must be sending kwargs anyway. Thus, what it does delegates correctly; its callers all send kwargs, it accepts kwargs, and forwards those kwargs on to real `find_by_sql` as kwargs.

If, hypothetically, `find_by_sql` had `ruby2_keywords` on it, then this would work:

```
    # Works as expected, passes preparable: true
    find_by_sql sql, [], { preparable: true }
```

Then, since such code might exist, we would _also_ need kasket's `find_by_sql` to accept either a trailing options hash, or keyword arguments. Putting `ruby2_keywords` on `find_by_sql_with_kasket` would then achieve that. As it is though, we don't need to do this because Rails' `find_by_sql` accepts kwargs exclusively.

ruby2_keywords is needed when you're blindly forwarding your arguments to a method, where you don't know whether that method takes kwargs or a trailing options hash. It works by marking a received last-position option hash with a special flag, that tells Ruby to convert it to kwargs _if_ calling a method that accepts kwargs.

In cases where you _know_ what you're forwarding to, and you _know_ that it takes kwargs, you can just delegate with `*args, **kwargs` (and if you _know_ it doesn't take kwargs, you can pass `kwargs` as a positional hash if needed too).

----

**ruby2_keywords is only needed when you don't know what you're forwarding to**

**Comment by HeyNonster** [comment]
> ```
>     # Won't raise an error, but will do entirely the wrong thing - passes the `{preparable: true}` hash as the `binds` parameter.
>     find_by_sql sql, { preparable: true }
> ```

I don't think this is true on Ruby 2.6, which we're still testing with:

```
def find_by_sql_without_kasket(sql, binds = [], preparable: nil, &block)
   puts "SQL: #{sql}"
   puts "Binds: #{binds}"
   puts "Preparable #{preparable}"
end

> RUBY_VERSION => "2.6.10"

> find_by_sql_without_kasket "foo", { preparable: true } =>

SQL: foo
Binds: []
Preparable true
```

**Comment by HeyNonster** [comment]
Also, the delegation changes the behavior on Ruby 2.6:

```
def delegate_to_find_by(sql, binds, *restargs, **kwargs, &blk)
  find_by_sql(sql, binds, *restargs, **kwargs, &blk)
end

delegate_to_find_by "foo", { preparable: true } =>

SQL: foo
Binds: {:preparable=>true}
Preparable: 
```

**Review by bquorning — APPROVED** [review]

**Pull request merged**